tests/PHPStan/Analyser/data/bug-7805.php

```diff
@@ -21,12 +21,10 @@ function foo(array $params)
 		assertNativeType("array<mixed~'help', mixed>", $params);
 		$params = $params === [] ? ['list'] : $params;
 		assertType("array{'list'}", $params);
-		assertNativeType("array{'list'}", $params);
-		// assertNativeType("array<mixed~'help', mixed>", $params); // not working yet
+		assertNativeType("non-empty-array<mixed~'help', mixed>", $params);
 		array_unshift($params, 'help');
 		assertType("array{'help', 'list'}", $params);
```

Comment:
shouldn't the `array_unshift($params, 'help');` also remove the `help` from the array shape (like it does for the native type)?